### PR TITLE
Add Match Route 12 - HF RRH Route

### PR DESCRIPTION
**Pull request by dtgreiner**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Add Match Route 12 as a new route to be used as the HF RRH Route

## Type of change
[//]: # 'remove options that are not relevant'
- [X] New feature (adds functionality)

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/mailers/route_twelve_mailer_methods.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      setup_instance_variables(notification)
+      mail(to: @contact.email, subject: "Match Declined by #{Translation.translate('Shelter Agency Ten')} - Requires Your Action")
+    end
+
+    def twelve_hsa_confirm_match_success(notification = nil)
+      setup_instance_variables(notification)
+      mail(to: @contact.email, subject: 'New Housing Recommendation - Requires Your Action')
+    end
+
+    def twelve_hsa_confirm_match_success_decline(notification = nil)
```
Comment (eanders) [inline app/mailers/route_twelve_mailer_methods.rb]: Should this be?
```suggestion
    def twelve_hsa_confirm_match_decline(notification = nil)
```

File: app/models/concerns/route_twelve_decisions.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+###
+
+module RouteTwelveDecisions
+  extend ActiveSupport::Concern
+
+  included do
+    has_decision :twelve_agency_acknowledges_receipt, decision_class_name: 'MatchDecisions::Twelve::TwelveAgencyAcknowledgesReceipt', notification_class_name: 'Notifications::Twelve::TwelveAgencyAcknowledgesReceipt'
+    has_decision :twelve_agency_acknowledges_receipt_decline, decision_class_name: 'MatchDecisions::Twelve::TwelveAgencyAcknowledgesReceiptDecline', notification_class_name: 'Notifications::Twelve::TwelveAgencyAcknowledgesReceiptDecline'
+    has_decision :twelve_hsa_confirm_match_success, decision_class_name: 'MatchDecisions::Twelve::TwelveHsaConfirmMatchSuccess', notification_class_name: 'Notifications::Twelve::TwelveHsaConfirmMatchSuccess'
+    has_decision :twelve_hsa_confirm_match_success_decline, decision_class_name: 'MatchDecisions::Twelve::TwelveHsaConfirmMatchSuccessDecline', notification_class_name: 'Notifications::Twelve::TwelveHsaConfirmMatchSuccessDecline'
```
Comment (eanders) [inline app/models/concerns/route_twelve_decisions.rb]: Pretty sure these shouldn't have the word Success in them:
```suggestion
    has_decision :twelve_hsa_confirm_match_decline, decision_class_name: 'MatchDecisions::Twelve::TwelveHsaConfirmMatchDecline', notification_class_name: 'Notifications::Twelve::TwelveHsaConfirmMatchDecline'
```

File: app/models/match_decisions/twelve/twelve_agency_acknowledges_receipt_decline.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    include MatchDecisions::AcceptsDeclineReason
+    include MatchDecisions::RouteTwelveDeclineReasons
+    include MatchDecisions::RouteTwelveCancelReasons
+
+    def to_partial_path
+      'match_decisions/twelve/agency_acknowledges_receipt_decline'
+    end
+
+    def step_name
+      "#{Translation.translate('CoC Twelve')} Confirms Match Success Decline"
```
Comment (eanders) [inline app/models/match_decisions/twelve/twelve_agency_acknowledges_receipt_decline.rb]: Is this step name right?  It doesn't match the model name.

File: app/models/match_decisions/twelve/twelve_hsa_confirm_match_success.rb [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    class StatusCallbacks < StatusCallbacks
+      def pending
+      end
+
+      def confirmed
+        @decision.next_step.initialize_decision!
+      end
+
+      def declined
+        match.twelve_hsa_confirm_match_success_decline_decision.initialize_decision!
```
Comment (eanders) [inline app/models/match_decisions/twelve/twelve_hsa_confirm_match_success.rb]: one more here

File: app/models/match_decisions/twelve/twelve_hsa_confirm_match_success_decline.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    include MatchDecisions::AcceptsDeclineReason
+    include MatchDecisions::RouteTwelveDeclineReasons
+    include MatchDecisions::RouteTwelveCancelReasons
+
+    def to_partial_path
+      'match_decisions/twelve/hsa_confirm_match_success_decline'
+    end
+
+    def step_name
+      "#{Translation.translate('CoC Twelve')} Confirms Match Success Decline"
```
Comment (eanders) [inline app/models/match_decisions/twelve/twelve_hsa_confirm_match_success_decline.rb]: Maybe **Match Success Decline** is the correct name for the model and methods, but this doesn't read well.  Happy to chat if you have thoughts.

Comment (eanders) [reply app/models/match_decisions/twelve/twelve_hsa_confirm_match_success_decline.rb]: Maybe **Match Success Review** is better?

**Review by eanders — COMMENTED** [review]
@dtgreiner this looks generally good, I'm a bit hung up on the "success decline" language (though it _does_ look like we use that in route 10 already.)

**Comment by dtgreiner** [comment]
> @dtgreiner this looks generally good, I'm a bit hung up on the "success decline" language (though it _does_ look like we use that in route 10 already.)

I wasn't a big fan of it either, but defaulted to what was there already. With that said, I don't see any problem with changing it up to be more clear.

File: app/models/match_decisions/twelve/twelve_hsa_confirm_match_success_decline.rb (hunk 5 above)
Comment (dtgreiner) [reply app/models/match_decisions/twelve/twelve_hsa_confirm_match_success_decline.rb]: I must have missed these step names when changing the success route names. I believe on this one they are confirming/overriding the decline, so maybe something like _Match Decline Review_?

Comment (eanders) [reply app/models/match_decisions/twelve/twelve_hsa_confirm_match_success_decline.rb]: That sounds reasonable!

**Comment by dtgreiner** [comment]
@eanders The success-decline language in the route should now be cleaned up

**Review by eanders — APPROVED** [review]
Thanks @dtgreiner !

**Pull request merged**